cli/connhelper/connhelper_test.go

```diff
@@ -63,3 +63,32 @@ func TestDisablePseudoTerminalAllocation(t *testing.T) {
 		})
 	}
 }
+
+func TestDockerSSHBinaryOverride(t *testing.T) {
+	testCases := []struct {
+		name     string
+		env      string
+		expected string
+	}{
+		{
+			name:     "Default",
+			env:      "",
+			expected: "docker",
+		},
+		{
+			name:     "Override",
+			env:      "other-binary",
+			expected: "other-binary",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Setenv(DockerSSHRemoteBinaryEnv, tc.env)
+			result := dockerSSHRemoteBinary()
+			if result != tc.expected {
+				t.Errorf("expected %q, got %q", tc.expected, result)
+			}
+		})
+	}
+}
```

Comment:
To be honest, the function tested here is so trivial, that it's in fact much simpler than the test code.
I don't think we really need it, but since it's already there, I guess it doesn't hurt either 🤷🏻